command/operator_raft_snapshot_save.go

```diff
@@ -71,24 +72,48 @@ func (c *OperatorRaftSnapshotSaveCommand) Run(args []string) int {
 		return 1
 	}
 
-	snapFile, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error opening output file: %s", err))
-		return 2
+	w := &lazyOpenWriter{
+		openFunc: func() (io.WriteCloser, error) {
+			return os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+		},
 	}
-	defer snapFile.Close()
+	defer w.Close()
 
 	client, err := c.Client()
 	if err != nil {
 		c.UI.Error(err.Error())
 		return 2
 	}
 
-	err = client.Sys().RaftSnapshot(snapFile)
+	err = client.Sys().RaftSnapshot(w)
 	if err != nil {
 		c.UI.Error(fmt.Sprintf("Error taking the snapshot: %s", err))
 		return 2
 	}
 
 	return 0
 }
+
+type lazyOpenWriter struct {
+	openFunc func() (io.WriteCloser, error)
+	writer io.WriteCloser
```

Comment:
It makes me a little uncomfortable that this isn't thread/data-race safe, but it looks like it won't break anything in the existing Sys.RaftSnapshot implementation.

Reply:
Yeah, my thought was this local, unexportable writer is only used in the one case.  If it were reusable I'd stick a mutex in it.